Add ClickSend notify service.

oozman opened the pull request:
The `clicksend` platform uses [ClickSend](https://clicksend.com) to deliver notifications from Home Assistant.

### Get your ClickSend API Credentials
Go to your [ClickSend Dashboard](https://dashboard.clicksend.com) section and create your new project. After creating your project, you should now be able to obtain your `username` and `api_key`.

### Checklist:

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated in home-assistant.github.io. See: [PR #2868](https://github.com/home-assistant/home-assistant.github.io/pull/2868)

If the code communicates with devices, web services, or third-party tools:

 - [x] Local tests with tox run successfully. Your PR cannot be merged unless tests pass
 - [x] New files were added to .coveragerc.

oozman committed:
Add ClickSend notify service.

homeassistant commented:
Hi @omarusman, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#8135).

Once you do that we will be able to review and accept this pull request.

Thanks!

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+"""
+Clicksend platform for notify component.
+For more details about this platform, please refer to the documentation at
+https://clicksend.com/help
```
Please point to the documentation at home-assistant.io.

omarusman replied:
Is this value fine: **https://home-assistant.io/docs/**
Because I haven't made a specific link for the documentation yet?

fabaff replied:
The link will be https://home-assistant.io/components/notify.clicksend/ for the platform documentation.

If the configuration is not passing the validation then a log entry is created which is composed on-the-fly and takes the platform name into account.

omarusman replied:
Thanks just made the changes.

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+# Set platform  parameters.
+CONF_API_URL = 'https://rest.clicksend.com/v3/sms/send'
+CONF_USERNAME = 'username'
+CONF_API_KEY = 'api_key'
```
Please use the constants defined in `const.py`.

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+def get_service(hass, config, discovery_info=None):
+
+    # Set notification service instance.
+    return ClicksendNotificationService(config[CONF_USERNAME], config[CONF_API_KEY], config[CONF_TO_NO])
```
Would be nice if you could check if the credentials are valid and make setup fails if not. Otherwise the users end-up with a non-working platform in their setup.

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+    """Implementation of a notification service for the Twitter service."""
+
+    def __init__(self, username, api_key, to_no):
+
```
Docstring is missing.

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+        self.to_no = to_no
+
+    def send_message(self, message="", **kwargs):
+
```
Dito

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+    def send_message(self, message="", **kwargs):
+
+        # Send request.
+        auth = self.username + ':' + self.api_key
```
Please use string formatting.

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+        auth = 'Basic ' + auth.decode('utf-8')
+
+        data = {'messages': [{'source': 'hass.notify', 'from': self.to_no, 'to': self.to_no, 'body': message}]}
+        headers = {'Content-type': 'application/json', 'Authorization': auth}
```
As far as I remember are there constants available for `Content-type` and `application/json`.

fabaff commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -0,0 +1,71 @@
+        data = {'messages': [{'source': 'hass.notify', 'from': self.to_no, 'to': self.to_no, 'body': message}]}
+        headers = {'Content-type': 'application/json', 'Authorization': auth}
+
+        resp = requests.post(CONF_API_URL, data=json.dumps(data), headers=headers)
```
Add a timeout to the request.

oozman committed:
PR #8135 changes.

- Some code spacing fixes.
- Add timeout to requests.
- Change doc url.
- Use const.py as much as possible.
- Check credentials to determine if setup fails or not.
- Add docstrings.
- Use string formatting.

oozman committed:
PR #8135 changes.

- Remove unused variables.
- Continuation line under-indented for visual indent.

oozman committed:
PR #8135 changes.

- Format code based on PEP8.

oozman committed:
PR #8135 changes.

- Remove unused base64 dependency.

oozman committed:
PR #8135 changes.

- Fix: D205: 1 blank line required between summary line and description (found 0)
- Fix: standard import "import json" comes before "import requests"

oozman committed:
PR #8135 changes.

- Add files to .coveragerc

fabaff committed:
Remove obvious comments and set constant
